apps/telegram_bot/main.py

```diff
@@ -0,0 +1,335 @@
+        )
+
+        # If we have a Coqui voice prompt, use that. Otherwise, set ID as synthesizer expects.
+        if voice_description is not None:
+            self.synthesizer.voice_prompt = voice_description
+        else:
+            setattr(self.synthesizer, voice_attr_of[type(self.synthesizer)], voice_id)
+
+        # Synthesize response
+        synth_response = await self.synthesizer.async_synthesize(agent_response)
+
+        # Save conversation to DB
+        return agent_response, synth_response
+
+    async def handle_telegram_start(
+        self, update: Update, context: ContextTypes.DEFAULT_TYPE
+    ):
+        chat_id = update.effective_chat.id
+        # Create user entry in DB
+        self.db[chat_id] = {
+            "voices": DEFAULT_VOICES,
+            "current_voice": DEFAULT_VOICES[0],
+            "current_conversation": None,
+        }
+        start_text = """
+I'm a voice chatbot, send a voice message to me and I'll send one back!" Use /help to see available commands.
+"""
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id, text=start_text
+        )
+
+    async def handle_telegram_message(
+        self, update: Update, context: ContextTypes.DEFAULT_TYPE
+    ):
+        chat_id = update.effective_chat.id
+        # Accept text or voice messages
+        if update.message.voice:
+            user_telegram_voice = await context.bot.get_file(
+                update.message.voice.file_id
+            )
+            bytes = await user_telegram_voice.download_as_bytearray()
+            input = AudioSegment.from_file(
+                io.BytesIO(bytes), format="ogg", codec="libopus"
+            )
+        elif update.message.text:
+            input = update.message.text
+        else:
+            # No audio or text, complain to user.
+            await context.bot.send_message(
+                chat_id=update.effective_chat.id,
+                text="""
+Sorry, I only respond to commands, voice, or text messages. Use /help for more information.""",
+            )
+            return
+
+        # Get audio response from LLM/synth and reply
+        agent_response, synth_response = await self.get_response(chat_id, input)
+        out_voice = io.BytesIO()
+        synth_response.export(out_f=out_voice, format="ogg", codec="libopus")
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id, text=agent_response
+        )
+        await context.bot.send_voice(chat_id=chat_id, voice=out_voice)
+
+    async def handle_telegram_select_voice(
+        self, update: Update, context: ContextTypes.DEFAULT_TYPE
+    ):
+        chat_id = update.effective_chat.id
+        if not (context.args):
+            await context.bot.send_message(
+                chat_id=chat_id,
+                text="You must include a voice id. Use /list to list available voices",
+            )
+            return
+        new_voice_id = context.args[0]
```

Comment:
do users select voice based on id or based on the voice name? ("obama"); should be based on the name, not id

if this is bc of the nameless coqui ones... we should have the user create names for them. something like

`/prompt-to-voice <voice name> <prompt>`